### PR TITLE
#90 #112 bridge and donation

### LINKED ISSUE #90: Fundmanager should transfer UBI to avatar's account on fuse
Definition of Done
- needs to know avatar's address on fuse
- needs to know bridge address
- should call correct methods to activate bridge (transferAndCall)

### LINKED ISSUE #112: Staking interest should be donated in first version
In first version we dont have a mechanism to cash out the GD interest paid back to the staking contract.
Staking contract should have a variable avgInterestDonationRatio
this variable in the current staking contract will be fixed at 100%
the method collectUBIInterest in staking model should return to caller also the donation ratio
fundmanager should then use donation ratio to tell reserve how much interest to pay back

### PR DESCRIPTION
**Pull request by sirpy**
# Description
- added avgInterestDonatedRatio 1e6 precission, currently set to 100%
- all minting goes to fundmanager
- fundmanager knows bridge and homeavatar (ie fuse/sidechain)
- fundmanager calls transferAndCall on bridge with avatar address as data
- fundmanager passes interest to mint to reserve after subtracting donation

About #90 #112 

TODO:
- fix unit tests as some names have changed and also minting logic and transfer amounts
- to test bridge use TransferAndCallMock style contract

### CODE REVIEW

File: stakingModel/test/GoodFundManager.test.js [hunk 1]
```diff
@@ -125,10 +127,12 @@ contract("GoodFundManager - transfer interest from the staking contract to the r
     let cdaiGains = gains["0"];
     let reserveCDaiBalanceBefore = await cDAI.balanceOf(goodReserve.address);
     let tx = await goodFundManager.transferInterest(simpleStaking.address);
+    let homeAvatarAfter = await goodDollar.balanceOf(homeAvatar);
     let reserveCDaiBalanceAfter = await cDAI.balanceOf(goodReserve.address);
     let stakingGDBalance = await goodDollar.balanceOf(simpleStaking.address);
     const gdPriceAfter = await marketMaker.currentPrice(cDAI.address);
-    expect(stakingGDBalance.toString()).to.be.equal("970492");
+    expect(stakingGDBalance.toString()).to.be.equal("0");
+    expect(homeAvatarAfter.toString()).to.be.equal("971086");
```
Comment (sirpy) [inline stakingModel/test/GoodFundManager.test.js]: hard coded numbers need to be explained
also why did the amount change from 970492 to 971086?

**Comment by sirpy** [comment]
@ordd resolve conflicts

File: stakingModel/test/GoodFundManager.test.js (hunk 1 above)
Comment (ordd) [reply stakingModel/test/GoodFundManager.test.js]: updated.
906867c66fb34ed7de644ef09ce6699fd747c0a8 

minting changes in reserve contract:
gdInterestMinted: 970492 (type: uint256),
gdExpansionMinted: 594 (type: uint256)

**Comment by ordd** [comment]
updated.
1bf4de5f4d1d0834a9265f42a6308ece04e2ee82

**Pull request merged**